kafka-client-examples/e2e-test/src/main/java/dev/responsive/examples/common/UncaughtStreamsAntithesisHandler.java

```diff
@@ -77,12 +78,11 @@ private boolean shouldLogError(final Throwable throwable, List<Throwable> seen)
         InjectedE2ETestException.class,
         InvalidProducerEpochException.class,
         MongoNotPrimaryException.class,
+        MongoSocketReadException.class,
+        MongoTimeoutException.class,
```

Comment:
I wonder if it might make sense to always inject a StreamsUncaughtExceptionHandler that checks for storage/Responsive-specific exceptions like this and opts to replace the thread if it sees a known exception. Obviously for anything else we delegate to the user's handler (if they have one) -- thoguhts?

Reply:
I think it's better to bubble them up - we should leverage the db client's native retry mechanisms instead of having two layers of retry mechanisms. Those clients have more fine-grained mechanisms (e.g. Cassandra knows what you can retry based on consistency guarantees, I assume Mongo has something similar).

The risk of just replacing the thread is that if there's actually a problem the backoff mechanisms won't be as effective if there's two layers of retries and we may exacerbate the issue.

Reply:
Yeah I tend to agree. I think we should do some level of retries at a level that's close to the actual error (e.g. by configuring the client or from the code that calls the client if the client doesn't support retries), and then just treat them like any other exception if they persist.